### PR TITLE
feat: default operator in getEntities of the sdk

### PR DESCRIPTION
**Pull request by cristianFleita**
## Introduced changes

Currently, the output query when retrieving multiple models using a composite query looks like this:
```typescript
{
  "Composite": {
    "operator": "And",
    "clauses": [
      {
        "Member": {
          "model": "jokers_of_neon-DeckCard",
          "member": "game_id",
          "operator": "Eq",
          "value": { "Primitive": { "U32": 3 } }
        }
      },
      {
        "Member": {
          "model": "jokers_of_neon-DeckCard",
          "member": "idx",
          "operator": "Gte",
          "value": { "Primitive": { "U32": 0 } }
        }
      },
      {
        "Member": {
          "model": "jokers_of_neon-Round",
          "member": "game_id",
          "operator": "Eq",
          "value": { "Primitive": { "U32": 3 } }
        }
      }
    ]
  }
}
```
The problem here is that the logical operator `And` is being used to combine the clauses, which only retrieves entities where both models (`DeckCard` and `Round`) exist together in the same query result. However, this is not always the desired behavior.

To address this, I introduced the ability to specify the logical operator (e.g., And, Or) in the getEntities function.

## Checklist

-   [ ] Linked relevant issue
-   [ ] Updated relevant documentation
-   [x] Added relevant tests
-   [ ] Add a dedicated CI job for new examples
-   [ ] Performed self-review of the code

## Summary by CodeRabbit

- **New Features**
	- Introduced a new parameter for specifying logical operators ("And" or "Or") in query functions, enhancing flexibility in query generation.
	- Added a test case for converting queries using the "Or" operator, expanding testing coverage.

- **Documentation**
	- Updated method signatures to reflect new parameters for better clarity.

### CODE REVIEW

File: packages/sdk/src/getEntities.ts [hunk 1]
```diff
@@ -36,9 +36,10 @@ export async function getEntities<T extends SchemaType>(
     }) => void,
     limit: number = 100, // Default limit
     offset: number = 0, // Default offset
-    options?: { logging?: boolean } // Logging option
+    options?: { logging?: boolean }, // Logging option
+    defaultOperator: "And" | "Or" = "And"
```
Comment (ponderingdemocritus) [inline packages/sdk/src/getEntities.ts:40]: this should be first in the params

File: packages/sdk/src/convertQuerytoClause.ts [hunk 2]
```diff
@@ -14,7 +14,8 @@ import { convertQueryToEntityKeyClauses } from "./convertQueryToEntityKeyClauses
  */
 export function convertQueryToClause<T extends SchemaType>(
     query: QueryType<T>,
-    schema: T
+    schema: T,
+    defaultOperator: "And" | "Or" = "And"
```
Comment (ponderingdemocritus) [inline packages/sdk/src/convertQuerytoClause.ts:18]: move above schema

**Review by ponderingdemocritus — COMMENTED** [review]
Looks good - can you update the typedoc above the functions with the changes.

**Comment by ponderingdemocritus** [comment]
Thinking about this more we should have the Operator within the Query itself, this makes the most sense. It is rather strange it is outside of the Query.

Can you make adjustments for this.

**Comment by cristianFleita** [comment]
> Thinking about this more we should have the Operator within the Query itself, this makes the most sense. It is rather strange it is outside of the Query.
> 
> Can you make adjustments for this.
> 
> It should exist within the Query Type
> 
> The functions should default to one or the other.
> 
> Use the types where possible.

Alright, that sounds better

**Comment by ponderingdemocritus** [comment]
closing as will open specific issue - i have made default OR. So to achieve your desire right now you can just use the default, or do a nested query.

**Pull request closed without merging**